site/src/pages/WorkspacePage/ResourcesSidebar.tsx

```diff
@@ -0,0 +1,109 @@
+import { Interpolation, Theme } from "@emotion/react";
+import Skeleton from "@mui/material/Skeleton";
+import { useTheme } from "@mui/material/styles";
+import { WorkspaceResource } from "api/typesGenerated";
+import {
+  Sidebar,
+  SidebarCaption,
+  SidebarItem,
+} from "components/FullPageLayout/Sidebar";
+import { getResourceIconPath } from "utils/workspace";
+
+type ResourcesSidebarProps = {
+  failed: boolean;
+  resources: WorkspaceResource[];
+  onChange: (resourceId: string) => void;
+  selected: string;
+};
+
+export const ResourcesSidebar = (props: ResourcesSidebarProps) => {
+  const theme = useTheme();
+  const { failed, onChange, selected, resources } = props;
+
+  return (
+    <Sidebar>
+      <SidebarCaption>Resources</SidebarCaption>
+      {failed && (
+        <p
+          css={{
+            margin: 0,
+            padding: "0 16px",
+            fontSize: 13,
+            color: theme.palette.text.secondary,
+            lineHeight: "1.5",
+          }}
+        >
+          Your workspace build failed, so the necessary resources couldn&apos;t
+          be created.
+        </p>
+      )}
+      {resources.length === 0 &&
+        !failed &&
+        Array.from({ length: 8 }, (_, i) => (
+          <SidebarItem key={i}>
+            <ResourceSidebarItemSkeleton />
+          </SidebarItem>
+        ))}
+      {resources.map((r) => (
+        <SidebarItem
+          onClick={() => onChange(r.id)}
+          isActive={r.id === selected}
+          key={r.id}
+          css={styles.root}
+        >
+          <div
+            css={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              lineHeight: 0,
```

Comment:
Do you want a lineHeight of 0, or a lineHeight of 1? I've had issues where a value of 0 broke a lot of text rendering and made text invisible the moment something accidentally overflowed to a new line

Reply:
I think 1 🤔 but let me check.

Reply:
Ahh in this case, it is an image wrapper so it is ok to use 0... I'm going to replace those wrappers - we have a few in the codebase - with the new @aslilac component `ExternalIcon`